cmake/Macros.cmake

```diff
@@ -35,6 +35,17 @@ function(sfml_set_stdlib target)
             message(FATAL_ERROR "Clang is the only supported compiler on macOS")
         endif()
     endif()
+
+    # on some platforms (e.g. ARM), GCC requires linking libatomic to use <atomic> features
+    if(SFML_COMPILER_GCC)
+        include(CheckCXXSourceCompiles)
+        check_cxx_source_compiles("#include <atomic>
+            #include <cstdint>
+            int main(){std::atomic<std::uint64_t> x(1); return x.fetch_add(1);}" ATOMIC_TEST)
+        if(NOT ATOMIC_TEST)
+            target_link_libraries(${target} PUBLIC "-latomic")
```

Comment:
Because we're only using this in Texture.cpp, can we link this as a `PRIVATE` dependency? I don't see why this needs to become a transitive dependency. Also, can't this merely be a link library for the graphics module only instead of a link library for every single module?

Reply:
We have to be careful here. Currently `<atomic>` is only used in Texture.cpp. Because it is likely that we might use `<atomic>` in more places as we modernize the code and replace stuff with `atomic<T>`s, I wouldn't want to have to copy and paste the same block of code into multiple places. This is a very unusual library because it is needed to support standard library features, as such I wouldn't put them in the same basket as the other libraries. Depending on whether we end up using `<atomic>` in a header somewhere, the library might also have to be made public. I can change it to private for now if you want, but there is a possibility we will have to make it public again some time in the future.

Reply:
> I wouldn't want to have to copy and paste the same block of code into multiple places.

You'll never need to do that. As soon as we need this in two module we'll put it in some higher common location.

> there is a possibility we will have to make it public again some time in the future.

This is perfectly fine. We'll make that change when the time comes.